be/src/formats/orc/orc_chunk_reader.cpp

```diff
@@ -1159,40 +159,97 @@ Status OrcChunkReader::_slot_to_orc_column_name(const SlotDescriptor* desc,
-    std::unordered_map<int, std::string> column_id_to_orc_name;
-    std::list<std::string> orc_column_names;
 
-    const auto& root_type = _reader->getType();
-    for (size_t i = 0; i < root_type.getSubtypeCount(); i++) {
-        const auto& sub_type = root_type.getSubtype(i);
-        column_id_to_orc_name.emplace(sub_type->getColumnId(), root_type.getFieldName(i));
-    }
+    std::list<uint64_t> include_column_id;
+    // Include root column id.
+    include_column_id.emplace_back(0);
 
-    for (SlotDescriptor* desc : _src_slot_descriptors) {
+    for (size_t i = 0; i < _src_slot_descriptors.size(); i++) {
+        SlotDescriptor* desc = _src_slot_descriptors[i];
         if (desc == nullptr) continue;
-        std::string orc_column_name;
-        RETURN_IF_ERROR(_slot_to_orc_column_name(desc, column_id_to_orc_name, &orc_column_name));
-        orc_column_names.emplace_back(orc_column_name);
+        RETURN_IF_ERROR(_add_include_column_id_by_slot(mapping, i, &include_column_id));
     }
 
-    _row_reader_options.include(orc_column_names);
+    _row_reader_options.includeTypes(include_column_id);
 
     if (_lazy_load_ctx != nullptr) {
-        std::list<std::string> orc_lazy_load_column_names;
-        for (SlotDescriptor* desc : _lazy_load_ctx->lazy_load_slots) {
+        std::list<uint64_t> lazy_load_column_id;
+
+        for (size_t pos : _lazy_load_ctx->lazy_load_indices) {
+            SlotDescriptor* desc = _src_slot_descriptors[pos];
             if (desc == nullptr) continue;
-            std::string orc_column_name;
-            RETURN_IF_ERROR(_slot_to_orc_column_name(desc, column_id_to_orc_name, &orc_column_name));
-            orc_lazy_load_column_names.emplace_back(orc_column_name);
+            RETURN_IF_ERROR(_add_include_column_id_by_slot(mapping, pos, &lazy_load_column_id));
         }
-        _row_reader_options.includeLazyLoadColumnNames(orc_lazy_load_column_names);
+
+        _row_reader_options.includeLazyLoadColumnIndexes(lazy_load_column_id);
     }
 
     return Status::OK();
 }
 
+Status OrcChunkReader::_add_include_column_id_by_slot(const std::unique_ptr<OrcMapping>& root_mapping,
+                                                      const uint64_t slot_pos, std::list<uint64_t>* column_id_list) {
+    const SlotDescriptor& desc = *_src_slot_descriptors[slot_pos];
+    column_id_list->push_back(root_mapping->get_column_id(slot_pos));
+
+    // Only complex type need include it's nested level.
+    if (desc.type().is_complex_type()) {
+        RETURN_IF_ERROR(
+                _add_include_column_id_by_type(desc.type(), root_mapping->get_child_mapping(slot_pos), column_id_list));
+    }
+    return Status::OK();
+}
+
+Status OrcChunkReader::_add_include_column_id_by_type(const TypeDescriptor& desc, const OrcMappingPtr mapping,
+                                                      std::list<uint64_t>* column_id_list) {
```

Comment:
```suggestion
Status OrcChunkReader::_add_include_column_id_by_type(const TypeDescriptor& desc, const OrcMappingPtr& mapping,
                                                      std::list<uint64_t>* column_id_list) {
```

Why do you use std::list rather than std::vector

Reply:
Because `apache-orc` is using `std::list<uint64_t>`.

We can't pass `const OrcMappingPtr&`, because mapping maybe be a nullptr.